Tests/kaas/k8s-default-storage-class/k8s-default-storage-class-check.py

```diff
@@ -180,28 +184,139 @@ def check_default_persistentvolumeclaim_readwriteonce(k8s_api_instance, storage_
                     "access mode 'ReadWriteOnce' is not supported",
                     return_code=42,
                 )
+    return 0
 
-    # 4. Delete resources used for testing
-    logger.debug(f"delete pod:{pod_name}")
-    api_response = k8s_api_instance.delete_namespaced_pod(pod_name, namespace)
-    logger.debug(f"delete pvc:{pvc_name}")
-    api_response = k8s_api_instance.delete_namespaced_persistent_volume_claim(
-        pvc_name, namespace
-    )
 
+def check_csi_provider(k8s_core_api, allowed_csi_prov=ALLOWED_CSI_PROV):
+    pods = k8s_core_api.list_namespaced_pod(namespace="kube-system")
+    csi_list = []
+    for pod in pods.items:
+        if "csi" in pod.metadata.name:
+            if any(provider in pod.metadata.name for provider in allowed_csi_prov):
+                csi_list.append(pod.metadata.name)
+                logger.info(f"CSI-Provider: {pod.metadata.name}")
+            else:
+                raise SCSTestException(
+                    f"CSI-Provider: {pod.metadata.name} not recommended",
+                    return_code=33,
+                )
+    if not csi_list:
+      logger.info("CSI-Provider: No CSI Provider found.")
```

Comment:
Why is this not an error? (I really don't know. Maybe a comment would help)
Also please note that the indentation is wrong here.

Reply:
its not an error, in case a kind cluster is used

Reply:
it should simply filter out "wrong" csi providers